### PR TITLE
fix issue #1127

### PR DESCRIPTION
**Pull request by 512222607**
// Describe your PR here; eg. Fixes #issueNo

Make sure that:

- [x] You have read the [contribution guidelines](https://dromara.org/en-us/docs/soul/contributor.html).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by yu199195** [comment]
hi， thanks， pls fix ci checkstyle

**Comment by yu199195** [comment]
pls ,  mvn clean install   in the local

**Comment by 512222607** [comment]
local clean install is ok

**Comment by 512222607** [comment]
image

File: soul-common/src/main/java/org/dromara/soul/common/dto/convert/ZombieUpstream.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package org.dromara.soul.common.dto.convert;
```
Comment (yu199195) [inline soul-common/src/main/java/org/dromara/soul/common/dto/convert/ZombieUpstream.java:1]: add apache license header

File: soul-admin/src/main/java/org/dromara/soul/admin/service/impl/UpstreamCheckService.java [hunk 2]
```diff
@@ -169,9 +215,15 @@ private void check(final String selectorName, final List<DivideUpstream> upstrea
...
                 divideUpstream.setStatus(false);
+                ZOMBIE_SET.add(ZombieUpstream.transform(divideUpstream, zombieCheckTimes, selectorName));
                 log.error("check the url={} is fail ", divideUpstream.getUpstreamUrl());
             }
         }
+        updateHandler(selectorName, upstreamList, successList);
+    }
+
+    private void updateHandler(final String selectorName, final List<DivideUpstream> upstreamList, final List<DivideUpstream> successList) {
+        //无节点变化，包括僵尸节点复活 及 活节点死亡
```
Comment (dengliming) [inline soul-admin/src/main/java/org/dromara/soul/admin/service/impl/UpstreamCheckService.java:226]: Please comment in English.

File: soul-admin/src/main/java/org/dromara/soul/admin/service/impl/UpstreamCheckService.java [hunk 3]
```diff
@@ -134,7 +146,13 @@ public static void removeByKey(final String selectorName) {
      */
     public void submit(final String selectorName, final DivideUpstream divideUpstream) {
         if (UPSTREAM_MAP.containsKey(selectorName)) {
-            UPSTREAM_MAP.get(selectorName).add(divideUpstream);
+            List<DivideUpstream> upstreams = UPSTREAM_MAP.getOrDefault(selectorName, Collections.emptyList());
+            Optional<DivideUpstream> exists = upstreams.stream().filter(item -> StringUtils.isNotBlank(item.getUpstreamUrl()) && item.getUpstreamUrl().equals(divideUpstream.getUpstreamUrl())).findFirst();
+            if (!exists.isPresent()) {
+                UPSTREAM_MAP.get(selectorName).add(divideUpstream);
```
Comment (dengliming) [inline soul-admin/src/main/java/org/dromara/soul/admin/service/impl/UpstreamCheckService.java:152]: Do we really need to call `UPSTREAM_MAP.get(selectorName)` again?

Comment (512222607) [reply soul-admin/src/main/java/org/dromara/soul/admin/service/impl/UpstreamCheckService.java:153]: this is to deal line too long.
the 152 is forgot to revise it.

Comment (512222607) [reply soul-admin/src/main/java/org/dromara/soul/admin/service/impl/UpstreamCheckService.java]: thanks

**Review by yu199195 — APPROVED** [review]

**Pull request merged**